core/code.py

```diff
@@ -33,8 +34,9 @@ def __init__(self, name, node, mem, cmd_ev):
         self.cmd_ev = cmd_ev
         self.mem = mem
 
-    def Call(self, pos_args, named_args):
-        # type: (List[value_t], Dict[str, value_t]) -> value_t
+    def Call(self, args):
+        # type: (typed_args.Reader) -> value_t
+        pos_args = args.RestPos()
```

Comment:
Hm now I realized typed_args.Reader() was mainly for builtin functions, to bridge mycpp static types with YSH dynamic types

In this case they're both dynamic.  But yeah when we do the full param/arg binding for funcs we can look at this, I think it will work fine